frb_codegen/src/utils/misc.rs

```diff
@@ -29,45 +104,42 @@ pub fn with_changed_file<F: FnOnce() -> crate::Result<()>>(
 ) -> crate::Result<()> {
     let content_original = fs::read_to_string(path)?;
     fs::write(path, content_original.clone() + append_content)?;
-
     f()?;
-
     Ok(fs::write(path, content_original)?)
 }
 
-pub fn find_all_duplicates<T>(iter: &[T]) -> Vec<T>
-where
-    T: Eq + Hash + Clone,
-{
-    let mut uniq = HashSet::new();
-    iter.iter()
-        .filter(|x| !uniq.insert(*x))
-        .cloned()
-        .collect::<Vec<_>>()
-}
-
 fn check_for_keywords(v: &[String]) -> anyhow::Result<()> {
     if let Some(s) = v.iter().find(|s| DART_KEYWORDS.contains(&s.as_str())) {
-        return Err(anyhow!("Api name cannot be a dart keyword: {}", s));
+        let err_msg = format!("Api name cannot be a dart keyword: {}", s);
+        log::warn!("{}", err_msg);
+        return Err(anyhow!(err_msg));
     };
     Ok(())
 }
 
-/// check api defined by users, if no duplicates, then generate all symbols (api function name),
-/// including those generated implicitly by frb
-pub fn get_symbols_if_no_duplicates(configs: &[crate::Opts]) -> Result<Vec<String>, anyhow::Error> {
+/// check duplication among regular defined API block(s), and return symbols in tuple of
+/// format `(all_no_shared_symbols, all_shared_symbols)`
+/// for `all_no_shared_symbols`: if there is duplication among EXPLICITLY defined APIs, it would panic;
+/// for `all_shared_symbols`: it would be extended if there is duplication among IMPLICITLY defined API,
+/// which should not exist in `all_no_shared_symbols`. While it is not empty, it means
+/// there are at least one API(symbol) is shared among regualr defined API blocks
+pub fn get_symbols_if_no_duplicates(
+    regular_configs: &[crate::Opts],
```

Comment:
nit: it would be great to use the type system to avoid mistakes like "pass a non-regular config into where it accepts regular config".

btw what is "non-regular" configs?